### DISCUSSION TITLE
Strategy matrix of arrays/objects

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [community/community#24981] [Actions]
<p>Is there a way to create this kind of matrix, essentially each matrix value to be an array:</p>
<pre><code class="lang-auto">strategy:
  matrix:
    arrays: [ ["one", "1"], ["two", "2"], ["three", "3"] ]
</code></pre>
<p>So that during each run, those two values would be available like this:</p>
<pre><code class="lang-auto">steps:
  - run: echo $NUMBER = $DIGIT
    env:
      NUMBER: ${{ matrix.arrays[0] }}
      DIGIT: ${{ matrix.arrays[1] }}
</code></pre>
<p>Or may be as key/value pairs:</p>
<pre><code class="lang-auto">strategy:
  matrix:
    arrays: [ {number: "one", digit: "1"}, {number: "two", digit: "2"}, {number: "three", digit: "3"} ]
steps:
  - run: echo $NUMBER = $DIGIT
    env:
      NUMBER: ${{ matrix.arrays.number }}
      DIGIT: ${{ matrix.arrays.digit }}
</code></pre>

**Comment by aleks-ivanov** [comment, marked as answer]
<p>UPDATE:</p>
<p>Turns out it can quite straight-forwardly be used in that manner. Just think of the matrix as a normal array:</p>
<pre><code class="lang-auto">strategy:
  matrix:
    arrays: [ 
      number: {name: "one", digit: "1"}, 
      number: {name: "two", digit: "2"}, 
      number: {name: "three", digit: "3"} 
    ]
steps:
  - run: echo "$NUMBER_NAME is $NUMBER_DIGIT" # output for each separate run is "one is 1", "two is 2" and "three is 3"
    env:
      NUMBER_NAME: ${{ matrix.arrays.number.name }}
      NUMBER_DIGIT: ${{ matrix.arrays.number.digit }}
</code></pre>
<p>Pretty cool!<br>
I do think this should be explicitly shown in the documentation :slight_smile:</p>
<p>UPDATE 2:</p>
<p>Well, I stand corrected :smiley:<br>
It works without making the object a property of the matrix, because by default the object is inherited by the matrix:</p>
<pre><code class="lang-auto">strategy:
  matrix:
    arrays: [ 
      {name: "one", digit: "1"}, 
      {name: "two", digit: "2"}, 
      {name: "three", digit: "3"} 
    ]
steps:
  - run: echo "$NUMBER_NAME is $NUMBER_DIGIT" # output for each separate run is "one is 1", "two is 2" and "three is 3"
    env:
      NUMBER_NAME: ${{ matrix.arrays.name }}
      NUMBER_DIGIT: ${{ matrix.arrays.digit }}
</code></pre>
<p>Notice how there is no <strong>number</strong> property in the matrix, nor the reference and the output is actually the same :slight_smile: Very nice!</p>

**Reply by leonnelleijing** [reply]
Hwo can we make exclude one in this case ?
